test/crontab/date_checker_test.exs

```diff
@@ -77,4 +78,27 @@ defmodule Crontab.DateCheckerTest do
       matches_date?(%Crontab.CronExpression{reboot: true}, base_date)
     end
   end
+
+  test "DST ambiguity checks correct" do
+    normal_date = DateTime.from_naive!(~N[2024-11-28 00:00:00], "Europe/Zurich")
+
+    {:ambiguous, date_earlier, date_later} =
+      DateTime.from_naive(~N[2024-10-27 02:30:00], "Europe/Zurich")
+
+    assert matches_date?(~e[* * * * * *]a, date_earlier)
+    refute matches_date?(~e[* * * * * *]a, date_later)
```

Comment:
@shaolang Oh yes, that makes sense. (Sorry, it has been a while and I need to wrap my head around it again fist...)

I however don't like that there's an inconsistency where a cron expression specifies behaviour later only, but it still evaluates truthy for both.

What do you think about doing that decision when initializing the cron expression struct?

It would work something like this:

**Modifiers**

(Changing the names earlier / later to previous / following since `e` is taken for extended.)

* `~e[...]b` (meaning `both`) -> `[:previous, :following]`
* `~e[...]p` -> `[:previous]`
* `~e[...]f` -> `[:following]`
* `~e[...]n` (meaning `neither) -> `[]`
* `~e[...]` -> autodetection

**Autodetection**

Analyzes the expression to see if it is an expression that should normally evaluate in `both` or `following` only.

As a condition for the default behavior, we could just check if the hour is constrained.

Reply:
I don't think the inconsistency is an issue: the user's intent is really to run the job every second/minute. `String.split` still works even when white space or the specified pattern doesn't exist in the given string. If you still feel strongly about this, I think a better way is to fail the parsing onset, e.g., `~e[*]al` will fail outright. I have no idea how much effort is needed if we were to adopt this route. Therefore, we have three options:
1. Stick with the "inconsistency"
2. Make second/minute expressions observe the same ambiguity rules.
3. Fail the parsing of such expressions.

As to the modifiers:
- Skipping ambiguous times can be done simply by not specifying any of the two modifiers, i.e., no need for `n`
- Autodetection isn't a good idea: we'd never be able to guess correctly for all scenarios; as people start filing issues, rules may evolve arbitrarily till it no longer make sense, yet the burden to maintain the code may become burdensome.
- I personally still don't think having a `b` shorthand for both times is a good idea, e.g., after serializing/deserializing, the cron expression may be `b` or `pf`; if we stick to the more "verbose" way, we'll always get back `pf`.